swagger/swagger-invocation/invocation-core/src/main/java/org/apache/servicecomb/swagger/invocation/context/InvocationContext.java

```diff
@@ -36,6 +38,11 @@
 
   protected Map<String, Object> localContext = new HashMap<>();
 
+  /**
+   * the vertx context that start the invocation
+   */
+  protected Context vertxContext;
```

Comment:
How about `T transportContext` ? Because vert.x or web container have different context.

Reply:
seems useless to record context for web container?  
and i don't know what is the context for web container, 😂

Reply:
How about this ?

```
Interface TransportContext

class VertxTransportContext implments TransportContext 
    {Context vertxContext; RoutingContext routingContext; ...}
class ServletTransportContext implements TransportContext
    {HttpServletRequest rquest; HttpServletResponse response;}
```

Reply:
request already can be got from invocation.  
and not want publish servlet response......

Reply:
maybe useful for new Filter mechanism, let me try it.

Reply:
OK. New filter mechanism should consider a 'global error handler'(errors thrown everywhere) .  This is a frequently asked questions and java chassis not handle it well now.

Reply:
yes, already design it.

Reply:
done